sdk/patina/src/perf_timer/component.rs

```diff
@@ -0,0 +1,112 @@
+//! Component that initializes the performance timer service.
+//! By default, it uses architecture-specified methods to determine the timer frequency.
+//! Consumers can override this by providing the `PerfFrequency` configuration.
+//!
+//! ## License
+//!
+//! Copyright (c) Microsoft Corporation.
+//!
+//! SPDX-License-Identifier: Apache-2.0
+//!
+//!
+
+use crate as patina;
+use crate::component::service::IntoService;
+use crate::{
+    component::{IntoComponent, params::*},
+    error::EfiError,
+    perf_timer::timer::ArchTimerFunctionality,
+};
+
+/// Performance frequency knob.
+/// Platforms can override this with a custom performance frequency;
+/// otherwise the default implementation will try to determine it via architecture specific methods.
+pub struct PerfFrequency(pub u64);
+
+impl PerfFrequency {
+    /// Create a new PerfFrequency with the given frequency in Hz.
+    pub fn new(frequency: u64) -> Self {
+        Self(frequency)
+    }
+}
+
+impl Default for PerfFrequency {
+    /// The default implementation tries to determine the frequency via architecture specific methods
+    /// if a platform does not provide their own configuration. (cpuid for x86, `CNTFRQ_EL0` for aarch64)
+    ///
+    /// However, if a platform registers their own configuration of this value, it will be used instead of this default.
+    fn default() -> Self {
+        let mut frequency = 0u64;
+
+        // Try to get TSC frequency from CPUID (most Intel and AMD platforms).
+        #[cfg(target_arch = "x86_64")]
```

Comment:
I'm not sure if I like putting more complex logic like this into a Default. Perhaps it would be better for this to have an internal Option, e.g. Option<u64>, and if none then the actual component will perform this logic. It just keeps the code more linear and explicit.